Code/Framework/AzCore/AzCore/std/hash_table.h

```diff
@@ -972,28 +987,32 @@ namespace AZStd
             rhs.clear();
         }
 
+        // find_insert_position sets `insertIter` to where the element should be inserted
+        // and returns true if the element should be inserted, otherwise false
         template<class ComparableToKey, class KeyEq>
-        bool find_insert_position(const ComparableToKey& keyCmp, const KeyEq& keyEq, iterator& iter, size_type numElements, const true_type& /* is multi elements */)
+        bool find_insert_position(const ComparableToKey& keyCmp, const KeyEq& keyEq, iterator& insertIter, size_type numElements, const true_type& /* is multi elements */)
         {
-            for (size_type i = 0; i < numElements; ++i, ++iter)
+            for (size_type i = 0; i < numElements; ++i, ++insertIter)
             {
-                if (keyEq(keyCmp, Traits::key_from_value(*iter)))
+                if (keyEq(keyCmp, Traits::key_from_value(*insertIter)))
                 {
-                    ++iter;
+                    ++insertIter;
                     break;
                 }
             }
 
+            // we always return true since multi elements (like multiset) allow to have repeated elements
```

Comment:
suggested: `// we always return true since multi elements (like multiset) allow to have repeated elements` -> 
`// always return true since multi elements (like multiset) allow repeated elements`